In [20]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

### **Merging all datasets into one**

Make sure to uncomment cells in this section in order to create a dataset that contains all data from data scraping.

In [21]:
# from os import listdir

# PATH = 'data_scraping/data/datasets/'

# # Concatenating all datasets
# for i, ds in enumerate(listdir(PATH)):
#     if i == 0:
#         data = pd.read_csv(f'{PATH}{ds}')
#     else:
#         data = pd.concat([data, pd.read_csv(f'{PATH}{ds}')])

# print(data.shape)
# data.head()

In [22]:
# # Translating column names from polish to english
# colnames = {
#     'Marka pojazdu': 'Brand',
#     'Model pojazdu': 'Model',
#     'Rok produkcji': 'Year',
#     'Przebieg': 'Mileage',
#     'Pojemność skokowa': 'Engine_Displacement',
#     'Rodzaj paliwa': 'Fuel_Type',
#     'Moc': 'Power',
#     'Skrzynia biegów': 'Transmission',
#     'Napęd': 'Drive',
#     'Spalanie W Mieście': 'City_Fuel_Consumption',
#     'Typ nadwozia': 'Body_Type',
#     'Liczba drzwi': 'Doors',
#     'Liczba miejsc': 'Seats',
#     'Kolor': 'Color',
#     'Kraj pochodzenia': 'Origin_Country',
#     'Bezwypadkowy': 'Accident_Free',
#     'Serwisowany w ASO': 'Serviced_at_ASO',
#     'Stan': 'Condition',
#     'Cena': 'Price'
# }
# data.rename(columns=colnames, inplace=True)

# data.head()

In [23]:
# # Save all data in one file
# data.to_csv('data_scraping/data/all_data.csv', index=False)

## **Data cleaning**

In [24]:
data = pd.read_csv('data_scraping/data/all_data.csv')
data.head()

Brand Model    Year     Mileage Engine_Displacement Fuel_Type   Power  \
0  Audi   SQ7  2017.0  142 800 km           3 956 cm3    Diesel  435 KM   
1  Audi    A3  2010.0  236 000 km           1 395 cm3   Benzyna  125 KM   
2  Audi    A4  2018.0   84 500 km           1 968 cm3    Diesel  190 KM   
3  Audi    Q5  2018.0   38 000 km           1 984 cm3   Benzyna  252 KM   
4  Audi    A7  2016.0   74 885 km           2 995 cm3   Benzyna  333 KM   

   Transmission             Drive City_Fuel_Consumption Body_Type  Doors  \
0  Automatyczna       4x4 (stały)             8 l/100km       SUV    5.0   
1      Manualna  Na przednie koła                   NaN   Kompakt    5.0   
2  Automatyczna       4x4 (stały)             5 l/100km     Kombi    5.0   
3  Automatyczna       4x4 (stały)             8 l/100km       SUV    5.0   
4  Automatyczna       4x4 (stały)            10 l/100km   Kompakt    5.0   

   Seats   Color Origin_Country Accident_Free Serviced_at_ASO Condition  \
0    7.0   Szary         Polska           Tak             Tak   Używane   
1    5.0  Czarny            NaN           Tak             Tak   Używane   
2    5.0   Szary            NaN           NaN             NaN   Używane   
3    5.0   Szary            NaN           NaN             NaN   Używane   
4    5.0   Biały            NaN           NaN             NaN   Używane   

                Price  
0  269 000        PLN  
1   29 900        PLN  
2   60 000        PLN  
3   75 000        PLN  
4   75 000        PLN

In [25]:
data.shape

(134057, 19)

This dataset comprises a vast collection of 134,057 observations specifically centered around cars. With 18 informative columns, it offers a diverse range of details pertaining to each vehicle. These encompass essential information such as the brand, model, production year, mileage, and fuel type. 

### **Missing values**

In [35]:
# How many missing values are there in each column
na_count = data.isna().sum()
na_precentage = (na_count / data.shape[0] * 100).round(2)

na_df = pd.DataFrame({'na_count': na_count, 'na_precentage': na_precentage})
na_df.sort_values(by='na_precentage', ascending=False)

na_count  na_precentage
Serviced_at_ASO           73520          54.84
Accident_Free             58608          43.72
City_Fuel_Consumption     54864          40.93
Origin_Country            45703          34.09
Drive                     17584          13.12
Seats                      8409           6.27
Engine_Displacement        2441           1.82
Mileage                    1093           0.82
Doors                       735           0.55
Power                        83           0.06
Transmission                 62           0.05
Color                         1           0.00
Condition                     0           0.00
Brand                         1           0.00
Body_Type                     1           0.00
Model                         1           0.00
Fuel_Type                     1           0.00
Year                          1           0.00
Price                         0           0.00

Majority of columns is missing come observations. Good information is that the columns that are missing 